config/config.go

```diff
@@ -0,0 +1,142 @@
+package config
+
+import (
+	"github.com/nhost/be/services/mimir/model"
+	"github.com/nhost/be/services/mimir/schema"
+	"github.com/nhost/cli/internal/generichelper"
+	"github.com/nhost/cli/internal/ports"
+	"github.com/nhost/cli/util"
+	"github.com/pelletier/go-toml/v2"
+)
+
+type Config struct {
+	c *model.ConfigConfig
+}
+
+func NewConfig(c *model.ConfigConfig) *Config {
+	return &Config{
+		c: c,
+	}
+}
+
+func (c Config) Validate() error {
+	s, err := schema.New()
+	if err != nil {
+		return err
+	}
+	return s.ValidateConfig(c.c)
+
+}
+
+func (c Config) Global() *model.ConfigGlobal {
+	return c.c.Global
+}
+
+func (c Config) Auth() *model.ConfigAuth {
+	return c.c.Auth
+}
+
+func (c Config) Provider() *model.ConfigProvider {
+	return c.c.Provider
+}
+
+func (c Config) Hasura() *model.ConfigHasura {
+	return c.c.Hasura
+}
+
+func (c Config) Functions() *model.ConfigFunctions {
+	return c.c.Functions
+}
+
+func (c Config) Storage() *model.ConfigStorage {
+	return c.c.Storage
+}
+
+func (c Config) Marshal() ([]byte, error) {
+	return toml.Marshal(c.c)
+}
+
+func DefaultConfig() (*Config, error) {
+	s, err := schema.New()
+	if err != nil {
+		return nil, err
+	}
+
+	c := &model.ConfigConfig{
+		Auth:     defaultAuthConfig(),
+		Provider: defaultProviderConfig(),
+		Hasura:   defaultHasuraConfig(),
+	}
+
+	c, err = s.Fill(c)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Config{c: c}, nil
+}
+
+func defaultAuthConfig() *model.ConfigAuth {
+	return &model.ConfigAuth{
+		Redirections: &model.ConfigAuthRedirections{
+			ClientUrl: generichelper.Pointerify("http://localhost:3000"),
+		},
+		Method: &model.ConfigAuthMethod{
```

Comment:
You don't need to pass those scopes, they are hardcoded in hasura-auth already. The scopes in the configuration are additive. And in the case they weren't they should be added to the schema then. Basically, you should be able to omit the entire `Method` attribute.

Reply:
I can move defaults to schema indeed, but first let's ask @elitan & @nunopato if I shall do it, the current defaults are - https://github.com/nhost/cli/blob/a97f0050f3871e2bd6ad698db01d4ff51d16a61f/nhost/nhost.go#L333-L507

Reply:
> And in the case they weren't they should be added to the schema then

I think this means we don't have to move the defaults to the schema. They are already being set by hasura auth. and if hasura auth was not setting them, the schema would be a better place to define them instead of hardcoding it here.

Reply:
updated